src/node/blockstorage.h

```diff
@@ -175,12 +188,18 @@ class BlockManager
     //! Check whether the block associated with this index entry is pruned or not.
     bool IsBlockPruned(const CBlockIndex* pblockindex) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
 
+    //! Create or update a prune lock identified by its name
+    void UpdatePruneLock(const std::string& name, const PruneLockInfo& lock_info) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
+
     ~BlockManager()
     {
         Unload();
     }
 };
 
+//! Find the first block that is not pruned
+const CBlockIndex* GetFirstStoredBlock(const CBlockIndex* start_block) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
Why is this a global function, when it could be a member of BlockManager? I think that'd make more sense, if we longer term want to remove the cs_main lock on storage stuff and give BlockManager its own mutex to take.

Reply:
> Why is this a global function, when it could be a member of BlockManager? I think that'd make more sense, if we longer term want to remove the cs_main lock on storage stuff and give BlockManager its own mutex to take.

Done in #24150.